lib/KDateRange/KDateInput.vue

```diff
@@ -5,17 +5,17 @@
       <KTextBox
         :ref="inputRef"
         :value="value"
-        type="text"
+        type="date"
         :label="legendText"
         autoComplete="off"
         :invalid="Boolean(errorMessage)"
         :showInvalidText="Boolean(errorMessage)"
         :invalidText="errorMessage"
+        :floatingLabel="false"
         @input="handleInput"
```

Comment:
Hrm, it would be preferable, rather than manually assigning event listeners here, to do the prevent default using vue event binding:
```
@click.prevent="() => {}"
@focus.prevent="() => {}"
```

See here for details: https://v2.vuejs.org/v2/guide/events.html#Event-Modifiers

But, as it seems that KTextBox is wrapping UiTextBox and neither is passing through `$listeners` to the `input` tag, then this will not work. So, nothing we can do here without a bigger refactor which is not desirable right now!